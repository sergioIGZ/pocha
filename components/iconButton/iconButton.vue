<template>
  <button
    @click="$emit('click', $event)"
    :class="`bg-${color}-500 hover:bg-${color}-700`"
  >
    <slot />
  </button>
</template>

<script>
// `bg-${color}-500 hover:bg-${color}-700 text-white font-bold py-2 px-4 rounded-full`
// :class="{
// 'bg-red-500': color === 'red',
//   'text-white font-bold py-2 px-4 rounded-full': true
// }"
export default {
  name: 'IconButton',
  props: {
    color: {
      type: String,
      required: false,
      default: 'blue'
    }
  }
}
</script>

<style scoped>
button {
  width: 30px;
  height: 30px;
  font-size: 0.5rem;
  display: flex;
  justify-content: center;
  align-items: center;
  @apply text-white font-bold py-2 px-4 rounded-full;
}
</style>
