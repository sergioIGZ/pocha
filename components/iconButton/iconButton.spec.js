import { shallowMount } from '@vue/test-utils'
import iconButton from './iconButton'

describe('iconButton Component', () => {
  describe('Behaviour', () => {
    it('should be mounted with no problems', () => {
      // Arrange.
      // Act.
      const wrapper = shallowMount(iconButton)
      // Assert.
      expect(wrapper).toMatchSnapshot()
    })
  })
})
