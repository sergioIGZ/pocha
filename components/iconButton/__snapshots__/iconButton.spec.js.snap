// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`iconButton Component Behaviour should be mounted with no problems 1`] = `<button class="bg-blue-500 hover:bg-blue-700"></button>`;
